src/components/ConfigInsights/index.tsx

```diff
@@ -39,6 +49,13 @@ function ConfigInsightsDetails({ configID }: Props) {
         `/api/db/config_analysis?config_id=eq.${configID}`
       );
       const data = (await res.json()) as ConfigTypeInsights[];
+      data.forEach((item) => {
```

Comment:
Just nitpicking here; we should avoid mutations like this, as they can introduce bugs that are very unpredictable and very hard to track (Javascript passes objects by reference, so you may be mutating something being used somewhere else). Instead, use the .map array function, which is the same number of lines but produces a new object.